File: src/main/java/kotlin_java/way2/CarFactory.java
package kotlin_java.way2;

public class CarFactory {

    public Car getCar(String model,
                      String licencePlate,
                      String manufacturer) {
        Car car = new Car();
        car.setModel(model);
        car.setLicencePlate(licencePlate);
        car.setManufacturer(manufacturer);
        return car;
    }
}

File: src/main/java/kotlin_java/way1/CarFactory.kt
package kotlin_java.way1

class CarFactory {

    fun getCar(model: String?,
               licencePlate: String?,
               manufacturer: String?): Car {
        val car = Car()
        car.model = model
        car.licencePlate = licencePlate
        car.manufacturer = manufacturer
        return car
    }
}